### PR TITLE
Remove all of the labels from the jawg map

### PR DESCRIPTION
**Pull request by brandesdavid**
## Describe the issue:

we need darkmode map, the sea colors should be changed, and because of the jawg map integrations, we have to include or own stations and names

## Explain how you solved the issue:
- added icons and station names
- added darkmode map
- added zoom levels for icons and names
- changed up the props to have line based icons

## Additional notes or considerations:
this is the first iteration of the darkmode. later, we can try to recreate the lines inside the vector lines/routes of the map, with station labels and more intuitive station name opacity levels

### CODE REVIEW

**Comment by brandesdavid** [comment]
> @brandesdavid Is this a branch from the darkmode branch? Or is this informally depending on darkmode?

no not really, i added the jawg dark mode map only

**Comment by johan-t** [comment]
> > @brandesdavid Is this a branch from the darkmode branch? Or is this informally depending on darkmode?
> 
> no not really, i added the jawg dark mode map only

@brandesdavid But why did you add the darkmode map in this pr? I am a bit confused

File: src/components/Map/MapLayers/StationLayer/StationLayer.tsx [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
     stations: StationGeoJSON;
 }
 
 const StationLayer: React.FC<StationLayerProps> = ({ stations }) => {
+    const map = useMap();
+
+    useEffect(() => {
+        const currentMap = map.current;
+        const image = new Image(16,16);
+        image.src = ubahn_icon; // replace with the path to your icon
```
Comment (johan-t) [inline src/components/Map/MapLayers/StationLayer/StationLayer.tsx:19]: remove comment

File: src/components/Map/MapLayers/StationLayer/StationLayer.tsx [hunk 2]
```diff
@@ -1,13 +1,35 @@
...
 interface StationLayerProps {
     stations: StationGeoJSON;
 }
 
 const StationLayer: React.FC<StationLayerProps> = ({ stations }) => {
+    const map = useMap();
+
+    useEffect(() => {
+        const currentMap = map.current;
+        const image = new Image(16,16);
```
Comment (johan-t) [inline src/components/Map/MapLayers/StationLayer/StationLayer.tsx:18]: I could set the icon to be a bit smaller maybe 12 x 12 but be weary of image looking like shit due to compression

File: src/components/Map/MapLayers/StationLayer/StationLayer.tsx [hunk 3]
```diff
@@ -1,13 +1,35 @@
...
+import sbahn_icon from '../../../../icons/sbahn.svg';
+
 interface StationLayerProps {
     stations: StationGeoJSON;
 }
 
 const StationLayer: React.FC<StationLayerProps> = ({ stations }) => {
+    const map = useMap();
+
+    useEffect(() => {
```
Comment (johan-t) [inline src/components/Map/MapLayers/StationLayer/StationLayer.tsx:16]: Maybe avoid code duplication by using a [factory](https://en.wikipedia.org/wiki/Factory_method_pattern)

File: src/components/Map/MapLayers/StationLayer/StationLayer.tsx [hunk 4]
```diff
@@ -1,13 +1,35 @@
...
+        const sbahn_image = new Image(16,16);
+        sbahn_image.src = sbahn_icon;
+        sbahn_image.onload = () => currentMap?.addImage('sbahn-icon', sbahn_image);
+
+        return () => {
+            currentMap?.removeImage('ubahn-icon');
+            currentMap?.removeImage('sbahn-icon');
+        };
+    }, [map]);
+    const priorityStations = ['Hauptbahnhof', 'Gesundbrunnen', 'Jungfernheide', 'Ostkreuz', 'Südkreuz', 'Westkreuz'];
```
Comment (johan-t) [inline src/components/Map/MapLayers/StationLayer/StationLayer.tsx:31]: Do you know what is up with this as it is not always visible on the same zoom level?

File: src/components/Map/MapLayers/StationLayer/StationLayer.tsx [hunk 5]
```diff
@@ -20,6 +42,87 @@ const StationLayer: React.FC<StationLayerProps> = ({ stations }) => {
...
+                    'icon-anchor': 'bottom',
+                    'text-offset': [0, 1],
+                }}
+                paint={{
+                    'text-color': '#fff',
+                    'text-halo-color': '#000',
+                    'text-halo-width': 1,
+
+                    // 'text-opacity': [
+                    //     'step',
```
Comment (johan-t) [inline src/components/Map/MapLayers/StationLayer/StationLayer.tsx:94]: Please remove the comment

File: src/components/Map/MapLayers/StationLayer/StationLayer.tsx [hunk 6]
```diff
@@ -20,6 +42,87 @@ const StationLayer: React.FC<StationLayerProps> = ({ stations }) => {
...
+                id='stationNameLayer'
+                type='symbol'
+                layout={{
+                    'text-field': ['get', 'name'],
+                    'text-size': 13,
+                    'icon-image': [
+                        'case',
+                        ['any',
+                            ['in', 'S1', ['get', 'lines']],
+                            ['in', 'S2', ['get', 'lines']],
```
Comment (johan-t) [inline src/components/Map/MapLayers/StationLayer/StationLayer.tsx:55]: I think some of the lines are missing such as S45, S46, S47 and maybe something more

**Review by johan-t — CHANGES_REQUESTED** [review]
Some smaller changes are needed. The map looks good

File: src/components/Map/MapLayers/StationLayer/StationLayer.tsx [hunk 7]
```diff
@@ -1,13 +1,40 @@
...
 
 import { StationGeoJSON } from '../../../../utils/types';
 
+import ubahn_icon from '../../../../icons/ubahn.svg';
+import sbahn_icon from '../../../../icons/sbahn.svg';
+
 interface StationLayerProps {
     stations: StationGeoJSON;
 }
+class IconFactory {
```
Comment (johan-t) [inline src/components/Map/MapLayers/StationLayer/StationLayer.tsx:12]: Beautiful

**Review by johan-t — APPROVED** [review]

**Comment by brandesdavid** [comment]
nice

**Pull request merged**